src/lib/socket/server.ts

```diff
@@ -45,4 +48,22 @@ export default class Manager {
       })
     );
   }
+
+  async onSocketSendMessage(this: io.Socket, data: any) {
+    await MessageService.create(data);
+    this.broadcast.emit('new_message', { message: data });
```

Comment:
Won't broadcast.emit() send this message to all clients connected to the server, not just those in the user's party. I might be wrong on this, but we will only want to send messages between users in the same party.